src/me.rs

```diff
@@ -159,14 +160,16 @@ pub fn motion_estimation(
       let blk_w = bsize.width();
       let blk_h = bsize.height();
       let (mvx_min, mvx_max, mvy_min, mvy_max) = get_mv_range(fi, bo, blk_w, blk_h);
-      let x_lo = po.x + ((-range + (pmv.col / 8) as isize).max(mvx_min / 8));
-      let x_hi = po.x + ((range + (pmv.col / 8) as isize).min(mvx_max / 8));
-      let y_lo = po.y + ((-range + (pmv.row / 8) as isize).max(mvy_min / 8));
-      let y_hi = po.y + ((range + (pmv.row / 8) as isize).min(mvy_max / 8));
+      let x_lo = po.x + ((-range + (cmv.col / 8) as isize).max(mvx_min / 8));
+      let x_hi = po.x + ((range + (cmv.col / 8) as isize).min(mvx_max / 8));
+      let y_lo = po.y + ((-range + (cmv.row / 8) as isize).max(mvy_min / 8));
+      let y_hi = po.y + ((range + (cmv.row / 8) as isize).min(mvy_max / 8));
 
-      let mut lowest_sad = 128 * 128 * 4096 as u32;
+      let mut lowest_cost = std::u32::MAX;
       let mut best_mv = MotionVector { row: 0, col: 0 };
 
+      let lambda = (get_lambda_sqrt(fi, bit_depth) * 128.0) as u32;
```

Comment:
Lambda is Q7 here?

Reply:
it's Q8, but a factor 0.5 is thrown in here. Maybe I could rewrite as 256 * 0.5 to clarify

Reply:
Oh, is the factor due to the x/y min step?

Reply:
0.5 is just a fudge factor. No deep theoretical connection here

Reply:
Oh, okay. Yeah, then I'd like it separated out / commented as that, just so it can be messed with later.